src/client/command_test.cc

```diff
@@ -190,8 +191,74 @@ TEST(CommandTest, FillFlags) {
   EXPECT_EQ("-cc1", *flags.other().begin());
   EXPECT_EQ(1, flags.compiler().plugins_size());
   EXPECT_EQ(plugin_name, flags.compiler().plugins(0).name());
+
+  // Check that macroses go to non-cached list.
+  EXPECT_NE(flags.non_cached().end(),
+            std::find(flags.non_cached().begin(), flags.non_cached().end(),
+                      "MACRO"));
```

Comment:
Tehically `-D` and `MACRO` are different flags. You can see an example here: http://rextester.com/LFEJ65675 (expand compiler flags).
So I believe it's enough for test to test just for `MACRO` existence in flags.
If you like, I could add two more expects: that there is also a `-D` flag, and that it precedes the `MACRO` flag.